src/utils/xml.ts

```diff
@@ -1,29 +1,40 @@
 import { XMLParser } from "fast-xml-parser"
 
-/**
- * Parses an XML string into a JavaScript object
- * @param xmlString The XML string to parse
- * @returns Parsed JavaScript object representation of the XML
- * @throws Error if the XML is invalid or parsing fails
- */
+const XML_ENTITIES: Record<string, string> = {
+	"&": "&amp;",
+	"<": "&lt;",
+	">": "&gt;",
+	'"': "&quot;",
+	"'": "&apos;",
+}
+
+export function decodeSelectedXmlEntities(str: string, targets: string[] = ["&"]): string {
+	let result = str
+
+	for (const char of targets) {
+		const entity = XML_ENTITIES[char]
+		if (entity) {
+			result = result.replace(new RegExp(entity, "g"), char)
+		}
+	}
+
+	return result
+}
+
 export function parseXml(xmlString: string, stopNodes?: string[]): unknown {
 	const _stopNodes = stopNodes ?? []
 	try {
 		const parser = new XMLParser({
-			// Preserve attribute types (don't convert numbers/booleans)
 			ignoreAttributes: false,
 			attributeNamePrefix: "@_",
-			// Parse numbers and booleans in text nodes
 			parseAttributeValue: true,
 			parseTagValue: true,
-			// Trim whitespace from text nodes
 			trimValues: true,
 			stopNodes: _stopNodes,
 		})
 
 		return parser.parse(xmlString)
 	} catch (error) {
-		// Enhance error message for better debugging
```

Comment:
Let's not make unnecessary changes like removing these comments, because that will make merge conflicts harder.